socket.go

```diff
@@ -0,0 +1,38 @@
+package shuttle
+
+import (
+	"log"
+	"net"
+	"time"
+)
+
+type Socket struct {
+	net.Listener
+	Shuttle *Shuttle
+}
+
+func NewSocket(s *Shuttle) *Socket {
+	l, err := net.Listen("unix", s.config.Socket)
+	if err != nil {
+		log.Fatalln(err)
+	}
+
+	return &Socket{
+		Listener: l,
+		Shuttle:  s,
+	}
+}
+
+func (k *Socket) Start() {
+	for {
+		c, err := k.Accept()
+		if err != nil {
+			if err, ok := err.(*net.OpError); ok && err.Timeout() {
+				continue
+			}
+			return
+		}
+		c.SetDeadline(time.Now().Add(1 * time.Second))
```

Comment:
See the comments for about deadlines here: http://golang.org/pkg/net/#Conn

Since ReadLogLines will close the socket whenever it encounters an error you are basically saying  that all IO must complete within 1 second. And so a user wouldn't be able to open the socket, write a line, wait 2 seconds and then write the next line w/o re-opening the socket. Is this the intended behavior?